### PR TITLE
Update nodeedit format, only store applied edits in Nexus log (SYN-6771)

### PR DESCRIPTION
**Pull request by Cisphyx**
- Use NID in nodeedits instead of BUID
- Remove conditional edits
- Hold Nexus lock while deconflicting edits and store them only in the Nexus log

### CODE REVIEW

File: synapse/lib/layer.py [hunk 1]
```diff
@@ -3401,22 +3454,24 @@ async def syncIndexEvents(self, offs, matchdef, wait=True):
...
                             or (etyp in (EDIT_PROP_SET, EDIT_PROP_DEL)
                                 and (vals[0] in propm or f'{form}:{vals[0]}' in propm))
                             or (etyp in (EDIT_TAG_SET, EDIT_TAG_DEL) and vals[0] in tagm)
                             or (etyp in (EDIT_TAGPROP_SET, EDIT_TAGPROP_DEL)
                                 and (vals[1] in tagpropm or f'{vals[0]}:{vals[1]}' in tagpropm))):
 
-                        yield (curoff, (buid, form, etyp, vals, meta))
+                        # This was returning conditional edits, not meta, but nothing
+                        # we have uses that
+                        yield (curoff, (nid, form, etyp, vals, ()))
```
Comment (Cisphyx) [inline synapse/lib/layer.py:3468]: This was returning conditional edits, not meta, but nothing we have uses that

File: synapse/lib/layer.py [hunk 2]
```diff
@@ -3401,22 +3470,22 @@ async def syncIndexEvents(self, offs, matchdef, wait=True):
...
+                for etyp, vals in edit:
                     if ((form in formm and etyp in (EDIT_NODE_ADD, EDIT_NODE_DEL))
                             or (etyp in (EDIT_PROP_SET, EDIT_PROP_DEL)
                                 and (vals[0] in propm or f'{form}:{vals[0]}' in propm))
                             or (etyp in (EDIT_TAG_SET, EDIT_TAG_DEL) and vals[0] in tagm)
                             or (etyp in (EDIT_TAGPROP_SET, EDIT_TAGPROP_DEL)
                                 and (vals[1] in tagpropm or f'{vals[0]}:{vals[1]}' in tagpropm))):
 
-                        yield (curoff, (buid, form, etyp, vals, meta))
+                        yield (curoff, (nid, form, etyp, vals))
```
Comment (Cisphyx) [inline synapse/lib/layer.py:3482]: This was returning conditional edits, not meta.

File: synapse/cortex.py [hunk 3]
```diff
@@ -2061,51 +2058,43 @@ def getNidNdef(self, nid):
...
             return nid
 
         nid = self.nextnid.to_bytes(8)
         self.nextnid += 1
 
         self.v3stor.put(nid, buid, db=self.nid2buid)
-        self.v3stor.put(nid, inc.to_bytes(8), db=self.nidrefs)
-
         self.v3stor.put(buid, nid, db=self.buid2nid)
+        self.v3stor.put(nid, s_msgpack.en(ndef), db=self.nid2ndef)
```
Comment (vEpiphyte) [inline synapse/cortex.py:2095]: think this needs to go above the `self.v3stor.put(buid, nid, db=self.buid2nid)` line

File: synapse/lib/stormtypes.py [hunk 4]
```diff
@@ -7381,28 +7388,24 @@ async def _methGetPropArrayCount(self, propname, valu=undef):
     @stormfunc(readonly=True)
     async def _methGetEdges(self, verb=None):
         verb = await toprim(verb)
-        todo = s_common.todo('getEdges', verb=verb)
-        async for edge in self.viewDynIter(todo, ('view', 'read')):
-            yield edge
+
+        viewiden = self.valu.get('iden')
+        self.runt.confirm(('view', 'read'), gateiden=viewiden)
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py:7393]: Is there a case where someone has lifted a node from a runtime and doesn't have view.read already?

**Pull request merged**